src/store/getters/state-getters.ts

```diff
@@ -367,8 +372,55 @@ export const isIncorporationAgreementValid = (state: StateIF): boolean => {
   return getIncorporationAgreementStep(state).valid
 }
 
-/** Whether all the incorporation steps are valid. */
+/** Is true when the step is valid. */
+export const isDefineDissolutionValid = (state: StateIF): boolean => {
+  return isTypeCoop(state)
+    ? getDissolutionStatementStep(state).valid &&
+      getHasCertificateDestroyed(state)
+    : true
+}
+
+/** Whether all the steps are valid. */
 export const isApplicationValid = (state: StateIF): boolean => {
+  if (isIncorporationFiling(state)) {
+    return isIncorporationApplicationValid(state)
+  } else if (isDissolutionFiling(state)) {
+    return isDissolutionValid(state)
+  }
+  return false
+}
+
+/** Whether all the dissolution steps are valid. */
+export const isDissolutionValid = (state: StateIF): boolean => {
+  const isStaff = isRoleStaff(state)
+
+  const isDocumentDeliveryValid = getDocumentDelivery(state).valid
+  const isCertifyValid = getCertifyState(state).valid && !!getCertifyState(state).certifiedBy
+
+  // Only for Staff role
+  const isCourtOrderValid = isStaff ? getCourtOrderStep(state).valid : true
+  const isStaffPaymentValid = isStaff ? getStaffPaymentStep(state).valid : true
+
+  const isEffectiveDateTimeValid = (
+    isBaseCompany(state) && isStaff
+      ? getEffectiveDateTime(state).valid
+      : true
+  )
+
+  return (
+    isDefineDissolutionValid(state) &&
+    isAffidavitValid(state) &&
+    isResolutionValid(state) &&
+    isDocumentDeliveryValid &&
+    isCertifyValid &&
+    isEffectiveDateTimeValid &&
+    isCourtOrderValid &&
+    isStaffPaymentValid
+  )
+}
```

Comment:
One thing to check here and perhaps @severinbeauvais could articulate it better, but we will need to verify Folio Number is Valid as well, although i believe this may only be for Staff filings at the moment??
If so, we may just need a fallback value until staff filings are built out, not sure.

Reply:
It depends on whose code goes in first.

I intend to submit a PR today with initial functionality (lots of little changes), and then have a second PR later with the rest of the functionality.

So probably, my first PR will go in, then this PR, then my second PR. Either way, between us, let's remember to include whatever Folio Number valid flag there is.

Reply:
I have submitted my initial FN changes, but as it turns out, I am still working on the validation part of it ... So I will add the FN validation as part of my ticket.

This conversation can therefore be resolved.

Reply:
I have submitted my initial FN changes, but as it turns out, I am still working on the validation part of it ... So I will add the FN validation as part of my ticket.

This conversation can therefore be resolved.